src/Assert/Assert.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Assert;
+
+use Wikimedia\Assert\Assert as WikimediaAssert;
+use Wikimedia\Assert\ParameterAssertionException;
+
+/**
+ * Provides functions to assure values meet certain preconditions relevant
+ * in Wikibase.
+ *
+ * @see Wikimedia\Assert\Assert
+ *
+ * @since 3.7
+ *
+ * @license GPL-2.0+
+ */
+class Assert {
```

Comment:
I'm not quite sure how I feel about giving this class such a generic name. I can see that its role is to contain methods that would be too specific for `Wikimedia\Assert` but it seems very procedural style to have a class like this with methods for very specific services that get used all over the codebase.